apps/web/src/query/goals.ts

```diff
@@ -0,0 +1,89 @@
+import {cache} from "react";
+import {Goal, GoalState} from "@/types/goals";
+import {getEnergySumForSensorInRange} from "@energyleaf/db/query";
+import {getDemoGoals} from "@/lib/demo/demo";
+import {getUserData} from "@/query/user";
+import {endOfDay, endOfMonth, endOfWeek, startOfDay, startOfMonth, startOfWeek} from "date-fns";
+
+export const getGoals = cache(async(userId: string, sensorId: string) => {
+    const dateNow = new Date();
+    const userData = (await getUserData(userId))?.user_data;
+    if (!userData) {
+        throw new Error("User data not found");
+    }
+
+    const userDailyLimit = userData.consumptionGoal;
+    if (!userDailyLimit) {
+        return [];
+    }
+
+    if (userId === "demo") {
+        return getDemoGoals(userDailyLimit);
+    }
+
+    return await Promise.all([
+        dailyGoal(sensorId, userDailyLimit, dateNow),
```

Comment:
Ich finde den Namen "Goal" irgendwie verwirrend, weil er ja sowohl den Ziel als auch den Ist-Wert enthält. Was hälst du vom Namen "GoalStatus"? Müsste man dann PR-weit anpassen
```suggestion
        getDailyGoalStatus(sensorId, userDailyLimit, dateNow),
```